src/pyrokinetics/equilibrium/eliteinp.py

```diff
@@ -0,0 +1,197 @@
+
+        # Units
+        len_units = units.meter
+        psi = data["Psi"]
+        npsi = len(psi)
+        ntheta = data["npol"]
+        F = data["fpol"] * units.tesla * units.meter
+        p = data["p"]
+        q = data["q"] * units.dimensionless
+
+        FF_prime = F * UnitSpline(psi, F)(psi, derivative=1)
+        p_prime = UnitSpline(psi, p)(psi, derivative=1)
+
+        # Flux surface contours
+        R2D = data["R"] * len_units
+        Z2D = data["z"] * len_units
+
+        # R_major can be obtained from the flux surfaces
+        R_major = (np.max(R2D, axis=1) + np.min(R2D, axis=1)) / 2
+        r_minor = (np.max(R2D, axis=1) - np.min(R2D, axis=1)) / 2
+        Z_mid = (np.max(Z2D, axis=1) + np.min(Z2D, axis=1)) / 2
+        a_minor = r_minor[-1]
+
+        # Flatten known grid and add axis once
+        surface_coords = np.stack((R2D[1:, :].m.ravel(), Z2D[1:, :].m.ravel()), -1)
+        surface_coords = np.append(
+            surface_coords, np.array([[R2D[0, 0].m, Z2D[0, 0].m]]), axis=0
+        )
+
+        surface_psi = np.repeat(psi[1:].magnitude, ntheta)
+        surface_psi = np.append(surface_psi, psi[0].m)
+
+        psi_interp = CloughTocher2DInterpolator(
+            surface_coords, surface_psi, fill_value=psi[-1].m * 1.1
+        )
+
+        R = np.linspace(min(R2D[-1, :]), max(R2D[-1, :]), npsi).m
+        Z = np.linspace(min(Z2D[-1, :]), max(Z2D[-1, :]), npsi).m
+        RZ_coords = np.stack([x.ravel() for x in np.meshgrid(R, Z)], -1)
+
+        psi_RZ = psi_interp(RZ_coords).reshape((npsi, npsi)).T * psi.units
+
+        # Magnetic field strength at axis
+        B_0 = F[0] / R_major[0]
+
+        # Infer total current
+        Ip = data.get("Ip")
+        if Ip is None:
+            try:
+                dpsi = np.gradient(psi.magnitude)
+                I_profile = (
+                    2
+                    * np.pi
+                    * np.mean(data["R"])
+                    * np.mean(data["Bt"])
+                    / (mu_0 * data["q"])
+                    * dpsi
+                )
+                Ip = np.trapezoid(I_profile, psi.magnitude)
+            except Exception as e:
+                raise IOError(f"Could not infer total current: {e}")
+
+        Ip *= units.ampere
+
+        return Equilibrium(
+            R=R,
+            Z=Z,
+            psi_RZ=psi_RZ,
+            psi=psi,
+            F=F,
+            FF_prime=FF_prime,
+            p=p,
+            p_prime=p_prime,
+            q=q,
+            R_major=R_major,
+            r_minor=r_minor,
+            Z_mid=Z_mid,
+            psi_lcfs=psi[-1],
+            a_minor=a_minor,
+            B_0=B_0,
+            I_p=Ip,
+            clockwise_phi=clockwise_phi,
+            cocos=cocos,
+            eq_type="ELITEINP",
+        )
+
+    def verify_file_type(self, filename: PathLike) -> None:
+        with open(filename, "r") as f:
+            head = f.read(50).upper()
+        if "HELENA GENERATED INPUT" not in head:
```

Comment:
Other codes _can_ write this format I think?

Reply:
SCENE writes `Input data for ELITE, shape =eqbm format` in the header. Maybe it is just a case of building up a set of valid header checks?

Reply:
I've added another small check for the HELENA output but looking at the SCENE output the format/variable names are different (`psi` instead of `Psi:`, `pprime` instead of `dp/dpsi`). This was ran like 10 years ago so idk if much has changed with SCENE but it seems unlikely

This could become a little convoluted if we aren't careful. Maybe we need a dictionary for each key for given source of eliteinp...

Reply:
I have updated the error messages so if someone wants a different format a useful message will pop up.